usr/src/uts/common/fs/zfs/sys/zap.h

```diff
@@ -87,22 +88,17 @@ extern "C" {
 
 /*
  * Specifies matching criteria for ZAP lookups.
+ * MT_EXACT - exact match, bypasses normalization to do strcmp()
+ * MT_FIRST - Normalizes the lookup - compose/decompose and then
+ *            match the normalized string.
+ * MT_CASE - Used to keep track of whether the lookup request is
+ *           case sensitive or not (i.e. Windows vs Unix.)
  */
 typedef enum matchtype
 {
-	/* Only find an exact match (non-normalized) */
-	MT_EXACT,
-	/*
-	 * If there is an exact match, find that, otherwise find the
-	 * first normalized match.
-	 */
-	MT_BEST,
-	/*
-	 * Find the "first" normalized (case and Unicode form) match;
-	 * the designated "first" match will not change as long as the
-	 * set of entries with this normalization doesn't change.
-	 */
-	MT_FIRST
+	MT_EXACT = 0x1,
+	MT_FIRST,
```

Comment:
On Mon, Mar 28, 2016 at 6:39 PM, Matthew Ahrens <notifications@github.com>
wrote:

> In usr/src/uts/common/fs/zfs/sys/zap.h
> <https://github.com/openzfs/openzfs/pull/77#discussion_r57645719>:
>
> >   */
> >  typedef enum matchtype
> >  {
> > -	/* Only find an exact match (non-normalized) */
> > -	MT_EXACT,
> > -	/*
> > -	 * If there is an exact match, find that, otherwise find the
> > -	 * first normalized match.
> > -	 */
> > -	MT_BEST,
> > -	/*
> > -	 * Find the "first" normalized (case and Unicode form) match;
> > -	 * the designated "first" match will not change as long as the
> > -	 * set of entries with this normalization doesn't change.
> > -	 */
> > -	MT_FIRST
> > +	MT_EXACT = 0x1,
> > +	MT_FIRST,
>
> If this is supposed to be a bitfield, then please explicitly set this to
> 0x2.
>

Agreed.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openzfs/openzfs/pull/77/files/713d9ff5d7adc49d9e5858a05121641625b6f8d3#r57645719>
>